modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -276,6 +302,13 @@ public void responseReady(NHttpServerConnection conn) {
                 response.start(conn);
                 conn.getContext().setAttribute(PassThroughConstants.RES_TO_CLIENT_WRITE_START_TIME,
                         System.currentTimeMillis());
+                if (sourceConfiguration.getCorrelationStatus()) {
+                    MDC.put(PassThroughConstants.CORRELATION_MDC_PROPERTY,
+                            conn.getContext().getAttribute(PassThroughConstants.CORRELATION_ID).toString());
+                    correlationLog.info(" | HTTP State | " + conn.getContext().
+                            getAttribute("http.connection") + " | RESPONSE WRITE STARTED");
+                    MDC.remove(PassThroughConstants.CORRELATION_MDC_PROPERTY);
+                }
                 metrics.incrementMessagesSent();
                 if (!response.hasEntity()) {
                    // Update stats as outputReady will not be triggered for no entity responses
```

Comment:
We can handle 202 accepted here, basically we need to do the same thing we are doing at encoder.isCompleted() of outputRead() here.